infrahub_sdk/store.py

```diff
@@ -20,130 +23,336 @@ def get_schema_name(schema: str | type[SchemaType] | None = None) -> str | None:
-        return None
+    def _get_by_internal_id(
+        self, internal_id: str, kind: str | None = None
+    ) -> InfrahubNode | InfrahubNodeSync | CoreNode | CoreNodeSync:
+        if internal_id not in self._objs:
+            raise NodeNotFoundError(
+                node_type="n/a",
+                identifier={"internal_id": [internal_id]},
+                message=f"Unable to find the node {internal_id!r} in the store ({self.branch_name})",
+            )
+
+        node = self._objs[internal_id]
+        if kind and kind not in node.get_all_kinds():
+            raise NodeInvalidError(
+                node_type=kind,
+                identifier={"internal_id": [internal_id]},
+                message=f"Found a node of kind {node.get_kind()} instead of {kind} for internal_id {internal_id!r} in the store ({self.branch_name})",
+            )
+
+        return node
+
+    def _get_by_key(
+        self, key: str, kind: str | None = None
+    ) -> InfrahubNode | InfrahubNodeSync | CoreNode | CoreNodeSync:
+        if key not in self._keys:
+            raise NodeNotFoundError(
+                node_type="n/a",
+                identifier={"key": [key]},
+                message=f"Unable to find the node {key!r} in the store ({self.branch_name})",
+            )
+
+        node = self._get_by_internal_id(self._keys[key])
+
+        if kind and node.get_kind() != kind:
+            raise NodeInvalidError(
+                node_type=kind,
+                identifier={"key": [key]},
+                message=f"Found a node of kind {node.get_kind()} instead of {kind} for key {key!r} in the store ({self.branch_name})",
+            )
+
+        return node
+
+    def _get_by_id(self, id: str, kind: str | None = None) -> InfrahubNode | InfrahubNodeSync | CoreNode | CoreNodeSync:
+        if id not in self._uuids:
+            raise NodeNotFoundError(
+                node_type="n/a",
+                identifier={"id": [id]},
+                message=f"Unable to find the node {id!r} in the store ({self.branch_name})",
+            )
+
+        node = self._get_by_internal_id(self._uuids[id])
+        if kind and kind not in node.get_all_kinds():
+            raise NodeInvalidError(
+                node_type=kind,
+                identifier={"id": [id]},
+                message=f"Found a node of kind {node.get_kind()} instead of {kind} for id {id!r} in the store ({self.branch_name})",
+            )
+
+        return node
+
+    def _get_by_hfid(
+        self, hfid: str, kind: str | None = None
+    ) -> InfrahubNode | InfrahubNodeSync | CoreNode | CoreNodeSync:
+        if not kind:
+            node_kind, node_hfid = parse_human_friendly_id(hfid)
+        elif kind and hfid.startswith(kind):
+            node_kind, node_hfid = parse_human_friendly_id(hfid)
+        else:
+            node_kind = kind
+            node_hfid = [hfid]
+
+        exception_to_raise_if_not_found = NodeNotFoundError(
+            node_type=node_kind or "unknown",
```

Comment:
We're mixing how we use node_type in the NodeNotFoundErrors here it's "unknown" where as in other places it's "n/a". Should we perhaps fix this within the error itself so that node_type is optional?